tests/e2e/staking/suite.go

```diff
@@ -70,15 +73,32 @@ func (s *E2ETestSuite) SetupSuite() {
 	s.Require().Equal(uint32(0), txRes.Code)
 	s.Require().NoError(s.network.WaitForNextBlock())
 
-	unbondingAmount := sdk.NewCoin(sdk.DefaultBondDenom, sdk.NewInt(5))
-
 	// unbonding the amount
-	out, err = MsgUnbondExec(val.ClientCtx, val.Address, val.ValAddress, unbondingAmount)
+	out, err = MsgUnbondExec(val.ClientCtx, val.Address, val.ValAddress, unbondCoin)
 	s.Require().NoError(err)
 	s.Require().NoError(val.ClientCtx.Codec.UnmarshalJSON(out.Bytes(), &txRes))
 	s.Require().Equal(uint32(0), txRes.Code)
 	s.Require().NoError(s.network.WaitForNextBlock())
 
+	// tokenize shares twice (once for the transfer and one for the redeem)
+	for i := 1; i <= 2; i++ {
+		MsgTokenizeSharesExec(
```

Comment:
Don't you need to do this? Otherwise you are checking against an err outside the loop, so the error will never change inside the loop?
```suggestion
		err := MsgTokenizeSharesExec(...)
				s.Require().NoError(err)
		s.Require().NoError(val.ClientCtx.Codec.UnmarshalJSON(out.Bytes(), &txRes))
		s.Require().Equal(uint32(0), txRes.Code)
		s.Require().NoError(s.network.WaitForNextBlock())
	}
```

Reply:
Agree. Good catch!